### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by wpmcgrath95**
I believe we could use either batter average so I chose the annual one. I don't think I needed to use any column transformations since I didn't have any null values but I used it to practice.

### CODE REVIEW

File: python_files/spark_with_baseball_data.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Using Spark With Baseball DB Assignment 3
+# Will McGrath
+# October 3, 2020
```
Comment (gramadin) [inline python_files/spark_with_baseball_data.py:3]: I like to add an est run time so others know if something is wrong when they run

**Review by gramadin — COMMENTED** [review]
Thank you for letting me join the group. i am already benefiting from getting to do PR reviews.

File: python_files/spark_with_baseball_data.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from pyspark.ml import Pipeline
+from pyspark.ml.feature import CountVectorizer
+from pyspark.sql import SparkSession
+from split_column_transform import SplitColumnTransform
+
+
+def main():
+    # Setup Spark and fixing time-zone issue
+    SUBMIT_ARGS = "--packages mysql:mysql-connector-java:8.0.11 pyspark-shell"
+    END_ARG = "?zeroDateTimeBehavior=CONVERT_TO_NULL&useUnicode=true& \
```
Comment (bwilder95) [inline python_files/spark_with_baseball_data.py:18]: Nice catch here as well, yep I had to troubleshoot something similar here to fix for null dates when I was setting things up.

File: python_files/spark_with_baseball_data.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        SELECT batter_counts.batter AS Batter,
+               REPLACE(YEAR(game.local_date),',','') AS The_Year,
+               COUNT(batter_counts.batter) AS Batter_Count,
+               SUM(batter_counts.Hit) AS Hit_Sum,
+               SUM(batter_counts.atBat) AS atBat_Sum,
+               SUM(batter_counts.Hit)/NULLIF(SUM(batter_counts.atBat), 0) AS Batting_AVG
+        FROM batter_counts
+        JOIN game ON batter_counts.game_id = game.game_id
+        GROUP BY Batter, The_Year
+        """
```
Comment (bwilder95) [inline python_files/spark_with_baseball_data.py:77]: ... I thought we had to do 100-day rolling average??

File: requirements.txt [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-#   
-#    pip-compile from pip-tools in requirements.dev.txt
+#
 #    pip-compile --output-file=requirements.txt requirements.in
 #
 joblib==0.16.0            # via scikit-learn
 numpy==1.18.4             # via -r requirements.in, pandas, scikit-learn, scipy
 pandas==1.1.1             # via -r requirements.in
 plotly==4.9.0             # via -r requirements.in
+py4j==0.10.9              # via pyspark
```
Comment (bwilder95) [inline requirements.txt:11]: What is this one for? I thought we just needed pyspark

**Review by bwilder95 — COMMENTED** [review]
Hi Will, my comments are light this week. I wasn't able to finish the assignment myself. But overall this looked great.

File: requirements.txt (hunk 4 above)
Comment (wpmcgrath95) [reply requirements.txt:11]: I have no idea haha it was added when I updated with the command.

File: python_files/spark_with_baseball_data.py (hunk 3 above)
Comment (wpmcgrath95) [reply python_files/spark_with_baseball_data.py:77]: I believe @dafrenchyman said we could choose either average

File: python_files/spark_with_baseball_data.py (hunk 2 above)
Comment (wpmcgrath95) [reply python_files/spark_with_baseball_data.py:18]: Yeah it was a pain setting up haha

File: python_files/spark_with_baseball_data.py (hunk 1 above)
Comment (wpmcgrath95) [reply python_files/spark_with_baseball_data.py:3]: Ok I might try that next time thanks!

File: python_files/spark_with_baseball_data.py (hunk 3 above)
Comment (dafrenchyman) [reply python_files/spark_with_baseball_data.py:77]: No,.. I'm pretty sure I wanted the 100 day rolling average.

Comment (wpmcgrath95) [reply python_files/spark_with_baseball_data.py:77]: Oops, I apologize I misunderstood! You are right @bwilder95.

**Review by dafrenchyman — APPROVED** [review]
Please review notes in canvas

**Pull request merged**